### PR TITLE
CIF-2289: create a v3 cif page

### PR DESCRIPTION
**Pull request by Buuhuu**
## Description

This PR creates a copy of the CIF v2 page inheriting from v3 of the WCM Core Components.

## Related Issue

CIF-2289

## Motivation and Context

Support of the v3 page features, in particular async clientlib loading.

## How Has This Been Tested?

Manually, using the existing e2e tests in Venia

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

**Review by herzog31 — APPROVED** [review]

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/catalogpage/v3/catalogpage/_cq_dialog/.content.xml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    xmlns:jcr="http://www.jcp.org/jcr/1.0" 
+    xmlns:nt="http://www.jcp.org/jcr/nt/1.0" 
+    xmlns:cq="http://www.day.com/jcr/cq/1.0" 
+    xmlns:granite="http://www.adobe.com/jcr/granite/1.0"
+    jcr:primaryType="nt:unstructured"
+    jcr:title="Page"
+    sling:resourceType="cq/gui/components/authoring/dialog"
+    extraClientlibs="[cq.common.wcm,core.wcm.components.page.v2.editor,cq.wcm.msm.properties,granite.contexthub.configuration,cq.siteadmin.admin.properties,cif.shell.picker]"
+    mode="edit"
+    trackingFeature="cif-core-components:catalogpage:v1">
```
Comment (LSantha) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/catalogpage/v3/catalogpage/_cq_dialog/.content.xml:12]: Wrong version?

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/page/v3/page/_cq_dialog/.content.xml [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    xmlns:jcr="http://www.jcp.org/jcr/1.0" 
+    xmlns:nt="http://www.jcp.org/jcr/nt/1.0" 
+    xmlns:cq="http://www.day.com/jcr/cq/1.0" 
+    xmlns:granite="http://www.adobe.com/jcr/granite/1.0"
+    jcr:primaryType="nt:unstructured"
+    jcr:title="Page"
+    sling:resourceType="cq/gui/components/authoring/dialog"
+    extraClientlibs="[cq.common.wcm,core.wcm.components.page.v2.editor,cq.wcm.msm.properties,granite.contexthub.configuration,cq.siteadmin.admin.properties,cif.shell.picker]"
+    mode="edit"
+    trackingFeature="cif-core-components:page:v2">
```
Comment (LSantha) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/page/v3/page/_cq_dialog/.content.xml:12]: v3?

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/catalogpage/v3/catalogpage/_cq_dialog/.content.xml (hunk 1 above)
Comment (Buuhuu) [reply ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/catalogpage/v3/catalogpage/_cq_dialog/.content.xml]: good catch!

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/page/v3/page/_cq_dialog/.content.xml (hunk 2 above)
Comment (Buuhuu) [reply ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/page/v3/page/_cq_dialog/.content.xml]: Good catch, I also updated the extra `core.wcm.components.page.v2.editor` clientlib to v3

**Review by LSantha — APPROVED** [review]

**Pull request merged**